store/tracekv/store.go

```diff
@@ -90,6 +91,25 @@ func (tkv *Store) ReverseIterator(start, end []byte) types.Iterator {
 	return tkv.iterator(start, end, false)
 }
 
+// GetAllKeysUsedInTrace reads through all traced operations and returns
+// a set of all the keys inside the trace operations
+func (tkv *Store) GetAllKeysUsedInTrace(buf bytes.Buffer) map[string]bool {
+	var traceOp traceOperation
+	var err error
+	keys := make(map[string]bool)
+	for {
+		traceOp, err = readOperation(&buf)
+		if err != nil {
+			return keys
+		}
```

Comment:
Are you sure you want to just return `keys` here if there's an error? Looking into the implementation of `readOperation`, there are several possible errors.

Reply:
added a check for the error we'd like to return on and panic otherwise